src/exp_log_geo.jl

```diff
@@ -57,6 +57,36 @@ geodesic(M::AbstractManifold, p, X) = t -> exp(M, p, X, t)
 geodesic(M::AbstractManifold, p, X, t::Real) = exp(M, p, X, t)
 geodesic(M::AbstractManifold, p, X, T::AbstractVector) = map(t -> exp(M, p, X, t), T)
 
+@doc raw"""
+    geodesic!(M::AbstractManifold, p, X) -> Function
+
+Get the geodesic with initial point `p` and velocity `X` on the [`AbstractManifold`](@ref) `M`.
+A geodesic is a curve of zero acceleration. That is for the curve ``γ_{p,X}: I → \mathcal M``,
+with ``γ_{p,X}(0) = p`` and ``\dot γ_{p,X}(0) = X`` a geodesic further fulfills
+
+```math
+∇_{\dot γ_{p,X}(t)} \dot γ_{p,X}(t) = 0,
+```
+
+i.e. the curve is acceleration free with respect to the Riemannian metric.
+This yields, that the curve has constant velocity that is locally distance-minimizing.
+
+This function returns a function `(q,t)` of (time) `t` that mutates `q``.
+
+    geodesic!(M::AbstractManifold, q, p, X, t::Real)
```

Comment:
I think it would be better to attach these docs to the respective methods below.

Reply:
Sure, I can do that tomorrow.